### PR TITLE
fix(mpp): validate TypeScript expires as RFC 3339

### PR DESCRIPTION
**Pull request by michael-moffett**
Supersedes #286 by @koriyoshi2041, carried here with him as co-author. His PR has an unsigned commit and has not moved since 14 August, and #282 now depends on this parser, so I'm picking it up as discussed.

Commit 1 is his diff unchanged. Commit 2 adds tests for the edge cases the parser decides that weren't covered: `+24:00`, `+00:60`, day `00`, month `13`, minute `60`, year `0000`, whitespace, `-00:00`.

Leap seconds: RFC 3339 §5.7 allows `23:59:60`; this parser now accepts it, clamps to `:59.999`, and only where a leap second can occur (end of a UTC month), matching the Rust SDK's behaviour.

Addresses the TypeScript issues in https://github.com/solana-foundation/pay-kit/issues/284. Unblocks #282.

### CODE REVIEW

File: typescript/packages/mpp/src/shared/rfc3339.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const offsetMinute = Number(offsetMinuteText ?? 0);
+
+    if (
+        month < 1 ||
+        month > 12 ||
+        day < 1 ||
+        day > daysInMonth(year, month) ||
+        hour > 23 ||
+        minute > 59 ||
+        second > 59 ||
```
[the remark this answers is not in the document]
Comment (michael-moffett) [reply typescript/packages/mpp/src/shared/rfc3339.ts]: Accepted and clamped in `ce0a09b`, same as the Rust SDK. Thanks.

**Comment by michael-moffett** [comment]
@EfeDurmaz16 #313 is up with the parser, author credited. This should unblock things.

**Review by koriyoshi2041 — APPROVED** [review]
Checked the carried parser and the leap-second follow-up at `ce0a09ba`. The implementation keeps the strict calendar/offset bounds from #286, and the UTC end-of-month check matches the Rust SDK boundary after offset normalization. Local receipts: 34/34 focused parser tests, package typecheck, Prettier, and diff hygiene all pass.

**Comment by EfeDurmaz16** [comment]
Thanks for picking this up! Could you change "Fixes #284" to "Addresses the TypeScript issues in #284"? That issue also tracks the other SDKs, so it should stay open after this merges.

**Comment by michael-moffett** [comment]
@EfeDurmaz16 Changed the description as requested. Thanks all for the quick look and feedback!

**Comment by michael-moffett** [comment]
Rebased onto main to pick up the Hono fix from #314 — same Audit failure as on #315. No changes to the parser
